src/dhtproto/client/NotifierTypes.d

```diff
@@ -89,3 +134,65 @@ public struct RequestKeyInfo
             this.request_id, this.key);
     }
 }
+
+import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+/*******************************************************************************
+
+    Mixin for method to serialize a record value.
+
+    Params:
+        dst = pointer to the buffer to be serialized into
+
+    TODO: deprecated, replace with implementation in swarm
+
+*******************************************************************************/
+
+template SerializeMethod ( alias dst )
+{
+    import ocean.util.serialize.contiguous.Contiguous;
+    import ocean.util.serialize.Version;
+    import ocean.util.serialize.contiguous.Serializer;
+    import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+    /***************************************************************************
+
+        Serializes `src` into `dst`, using a version decorater, if required.
+
+        Params:
+            T = type of struct to serialize
+            src = instance to serialize
+
+    ***************************************************************************/
+
+    public void serialize ( T ) ( T src )
+    {
+        static if ( Version.Info!(T).exists )
+        {
+            if ( client_serializer_version_decorator is null )
+                client_serializer_version_decorator = new VersionDecorator;
+
+            client_serializer_version_decorator.store!(T)(src, *dst);
+        }
+        else
+        {
+            Serializer.serialize(src, *dst);
+        }
+    }
+}
+
+/*******************************************************************************
+
+    VersionDecorator singleton used by all mixed-in serialize() methods when
+    handling versioned structs.
+
+    Must be public as it needs to be accessible from the point where the
+    serialize method is mixed in. Not intended to be used externally by users.
+    The object is given an unusually long name so as to reduce the likelihood of
+    name clashes with other symbols.
+
+    TODO: deprecated, replace with implementation in swarm
+
+*******************************************************************************/
+
+public VersionDecorator client_serializer_version_decorator;
```

Comment:
In line with remarks above about `VersionDecorator.store`, this should not be necessary, so you don't even need to bother to deprecate it ;-)